### PR TITLE
[LV] Fix missing precomptueCosts() in emitInvalidCostRemarks().

### PR DESCRIPTION
**Pull request by ElvisWang123**
We should always update the `SkipComputation` which is set in `VPCostContext` before VPlan compute costs.

This patch prevent the assertion of in-loop reduction in the `VPReductionRecipe::computeCost()` and other potential assertions of partially implemented VPlan-based cost model.

### CODE REVIEW

**Comment by ElvisWang123** [comment]
Fail example: https://godbolt.org/z/96qha7hh8

File: llvm/test/Transforms/LoopVectorize/reduction-inloop.ll [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
 ; RUN: opt < %s -passes=loop-vectorize,dce,instcombine -force-vector-interleave=1 -force-vector-width=4 -prefer-inloop-reductions -S | FileCheck %s
+; Check remark analysis runs successfully.
+; RUN: opt < %s -passes=loop-vectorize,dce,instcombine -force-vector-interleave=1 -force-vector-width=4 -prefer-inloop-reductions -pass-remarks-analysis=loop-vectorize -S
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/reduction-inloop.ll:4]: Might be better to have a separate test case that also checks the output of the remark as well? Also no need to run `dce,instcombine`.

Comment (ElvisWang123) [reply llvm/test/Transforms/LoopVectorize/reduction-inloop.ll]: Add new test case for this failure, thanks.

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**